### DISCUSSION TITLE
Energy error increase a lot when add force label

### DISCUSSION THREAD

**Discussion by plumbum082** [deepmodeling/deepmd-kit#1996] [Q&A]
Hi, I meet a problem, when I train DP model with only energy label, the energy rmse error is about 1e-3 eV/Å ,however, when I add the force label ("start_pref_e": 1, "limit_pref_e": 1, "start_pref_f": 1, "limit_pref_f": 1), the force rmse error looks okay, nearly the same with the case using only force label, that is about 1e-2 eV/Å,  while the energy error increase to 1e-2eV/Å and the pair plot of energy looks not reasonable at all.  Could you help me?

below is the case with both energy and force label：
figures
![Figure_2](https://user-images.githubusercontent.com/54125554/195490938-b5fba9d3-ea8c-47be-b4b9-b6b5a16edf82.png)
![Figure_1](https://user-images.githubusercontent.com/54125554/195490847-a96f7090-af5d-45fb-8375-4054ef8b62e3.png)

input.json
{
    "_comment": " model parameters",
    "model": {
        "type_map":     ["O", "H"],
        "descriptor" :{
            "type":             "se_e2_a",
            "sel":              [20, 40],
            "rcut_smth":        3.8,
            "rcut":             4.0,
            "neuron":           [10, 20, 40, 80, 160],
            "resnet_dt":        false,
            "axis_neuron":      16,
            "seed":             1,
            "_comment":         " that's all"
        },
        "fitting_net" : {
            "neuron":           [160, 80, 40, 20],
            "resnet_dt":        true,
            "seed":             1,
            "_comment":         " that's all"
        },
        "_comment":     " that's all"
    },

    "learning_rate" :{
        "type":         "exp",
        "decay_steps":  1000,
        "start_lr":     0.001,
        "stop_lr":      3.51e-7,
        "_comment":     "that's all"
    },

    "loss" :{
        "type":         "ener",
        "start_pref_e": 1,
        "limit_pref_e": 1,
        "start_pref_f": 1,
        "limit_pref_f": 1,
        "start_pref_v": 0,
        "limit_pref_v": 0,
        "_comment":     " that's all"
    },

    "training" : {
        "training_data": {
            "systems":          ["../data_splt/dimer_t/", "../data_splt/trimer_t/", "../data_splt/4mer_t/", "../data_splt/5mer_t/", "../data_splt/4k_t/"],
            "batch_size":       "auto",
            "_comment":         "that's all"
        },
        "validation_data":{
            "systems":          ["../data_splt/dimer_v/", "../data_splt/trimer_v/", "../data_splt/4mer_v/", "../data_splt/5mer_v/", "../data_splt/4k_v/"],
            "batch_size":       "auto",
            "numb_btch":        5,
            "_comment":         "that's all"
        },
        "numb_steps":   600000,
        "seed":         10,
        "disp_file":    "lcurve.out",
        "disp_freq":    500,
        "save_freq":    1000,
        "_comment":     "that's all"
    },

    "_comment":         "that's all"
}

**Comment by njzjz** [comment]
I see your `rcut` is only 4 Å so it may discard interaction out of 4 Å.

**Reply by plumbum082** [reply]
Thank you for your reply! Yes, 4 Å is the setting of our model.  Sorry, I could not understand what you mean, could you explain it?  Besides, the energy and force labels are some terms of the interaction energy and force labels calculated by the difference of the values computed by different methods, will that have some influence ?

**Reply by njzjz** [reply]
In the model, $\frac{\partial E}{\partial r_{ij}}=0,~r_{ij}>r_{\text{cutoff}}$, meaning the interaction out of the cutoff radius cannot be fitted. If you use the difference between two different potentials, it's better to ensure their long-range parts are similar.

In addition, it looks that your black curve is not y=x... Can you confirm it?
![image](https://user-images.githubusercontent.com/9496702/195909055-54b2c23b-527c-4307-8e4f-74a03c1f6c33.png)

**Reply by plumbum082** [reply]
Okay, Thank you very much! Then should I try a larger rcut?  My apologies, there is a bug in my code for the graph.

**Comment by njzjz** [comment]
By the way, eV/A is a force unit. Do you mean eV when you use it as the energy unit?

**Reply by plumbum082** [reply]
oh, I am sorry, when I copy and paste I forgot to make a change. Yes, I mean eV when I use it as the energy unit.
